app/graphql/types/git/directory_entry_type.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+Types::Git::DirectoryEntryType = GraphQL::ObjectType.define do
```

Comment:
This could be an interface type with `FileType` being one member and adding `DirectoryType` with the fields `name` and `path`. Then `type` could be removed in favour of `__typename`.